Correctly handle ``__tracebackhide__`` for chained exceptions.
